server/join_test.go

```diff
@@ -14,29 +14,37 @@
 package server
 
 import (
+	"fmt"
 	"math/rand"
-	"strings"
-	"testing"
+	"os"
+	"sync"
 	"time"
 
-	"github.com/coreos/etcd/clientv3"
 	"github.com/juju/errors"
 	. "github.com/pingcap/check"
 	"golang.org/x/net/context"
 )
 
-func TestJoin(t *testing.T) {
-	TestingT(t)
-}
-
 var _ = Suite(&testJoinServerSuite{})
 
+var (
+	errTimeout = errors.New("timeout")
+)
+
 type testJoinServerSuite struct {
-	cfgs []*Config
+	m    sync.Mutex
+	dirs []string
 }
 
-func (s *testJoinServerSuite) SetUpSuite(c *C) {
-	s.cfgs = newTestMultiJoinConfig(3)
+func (s *testJoinServerSuite) TearDownSuite(c *C) {
+	// wait a while, avoid `etcdserver: failed to purge snap file`
+	time.Sleep(5 * time.Second)
```

Comment:
why sleep and lock?
In TearDown, we must ensure all servers are closed, so we can remove dir directly.